pkg/distributor/distributor.go

```diff
@@ -818,12 +819,19 @@ func (d *Distributor) prePushRelabelMiddleware(next push.Func) push.Func {
 			ts := req.Timeseries[tsIdx]
 
 			if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
-				l, keep := relabel.Process(mimirpb.FromLabelAdaptersToLabels(ts.Labels), mrc...)
+				lb := labels.NewBuilder(mimirpb.FromLabelAdaptersToLabels(ts.Labels))
```

Comment:
We could reuse the builder between timeseries, to avoid creating new one in each iteration.

Reply:
Let's see if I understand you right)